addons/stock/models/stock_quant.py

```diff
@@ -341,6 +429,46 @@ def _quant_tasks(self):
         self._merge_quants()
         self._unlink_zero_quants()
 
+    def write(self, vals):
+        if 'inventory_quantity' in vals and not self.env.context.get('dont_create_inventory_line'):
+            for quant in self:
+                rounding = quant.product_id.uom_id.rounding
+                diff = float_round(vals['inventory_quantity'] - quant.quantity, precision_rounding=rounding)
+                diff_float_compared = float_compare(diff, 0, precision_rounding=rounding)
+                if diff_float_compared == 0:
+                    continue
+                elif diff_float_compared > 0:
+                    move = self.env['stock.move'].create(self._get_inventory_move_values(self.product_id.property_stock_inventory, self.location_id, diff))
+                    move._action_done()
+                else:
+                    move = self.env['stock.move'].create(self._get_inventory_move_values(self.location_id, self.product_id.property_stock_inventory, -1 * diff))
+                    move._action_done()
+            vals.pop('inventory_quantity')
+        return super(StockQuant, self).write(vals)
+
+    @api.model
+    def create(self, vals):
+        if 'inventory_quantity' in vals and not self.env.context.get('dont_create_inventory_line'):
+            if 'product_id' in vals:
+                product = self.env['product.product'].browse(vals['product_id'])
+            else:
+                product = self._default_product_id()
```

Comment:
in which case don't you have a product here? not sure it is safe to fallback on the default one..

Reply:
I changed that with a `force_save`. Thanks !